openfl/federated/data/sources/data_source.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Base class for different types of data sources."""
+
+from abc import ABC, abstractmethod
+from enum import Enum
+
+
+class DataSourceType(Enum):
+    """Enum for the different types of data sources."""
+
+    LOCAL = "local"
+    S3 = "s3"
+
+
+class DataSource(ABC):
```

Comment:
I'd suggest adding some documentation and examples on how the dataset/data source abstractions can be used in practice under the [OpenFL Utilities](https://openfl.readthedocs.io/en/latest/developer_guide/utilities.html) section of the developer guide. This could be a separate `.rst` document plugged in here:
https://github.com/securefederatedai/openfl/tree/develop/docs/developer_guide/utilities

PS: this could be done in a separate PR